codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -66,32 +57,19 @@
  * free to close the channel in response to such events if they don't have use for any queued
  * messages.
  *
- * <p>{@link ChannelConfig#setMaxMessagesPerRead(int)} and {@link
- * ChannelConfig#setAutoRead(boolean)} are supported.
+ * <p>{@link ChannelConfig#setMaxMessagesPerRead(int)} and {@link ChannelConfig#setAutoRead(boolean)} are supported.
  */
 @UnstableApi
 public final class Http2MultiplexCodec extends ChannelDuplexHandler {
-    private static final Http2FrameLogger HTTP2_FRAME_LOGGER = new Http2FrameLogger(INFO, Http2MultiplexCodec.class);
 
     private final ChannelHandler streamHandler;
     private final EventLoopGroup streamGroup;
-    private final Http2ConnectionHandler http2Handler;
-    private final Http2Connection.PropertyKey streamInfoKey;
-    private final List<StreamInfo> streamsToFireChildReadComplete = new ArrayList<StreamInfo>();
+    private final List<Http2StreamChannel> channelsToFireChildReadComplete = new ArrayList<Http2StreamChannel>();
+    private final boolean server;
     private ChannelHandlerContext ctx;
-    private ChannelHandlerContext http2HandlerCtx;
     private volatile Runnable flushTask;
 
-    /**
-     * Construct a new handler whose child channels run in the same event loop as this handler.
-     *
-     * @param server {@code true} this is a server
-     * @param streamHandler the handler added to channels for remotely-created streams. It must be
-     *     {@link ChannelHandler.Sharable}.
-     */
-    public Http2MultiplexCodec(boolean server, ChannelHandler streamHandler) {
-        this(server, streamHandler, null);
-    }
+    private final IntObjectMap<Http2StreamChannel> childChannels = new IntObjectHashMap<Http2StreamChannel>();
```

Comment:
@normanmaurer no. it's the number of active HTTP/2 streams. it could really be anything. the default initial size is 8.